app/services/aws/credentials_service.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+module Aws
+  class CredentialsService
+    class << self
+      def call(params = {})
+        new(params).call
+      end
+    end
+
+    class AuthenticationTypeError < StandardError
+    end
+
+    IAM_KEYS = %i[access_key_id secret_access_key].freeze
+    STS_KEYS = %i[region role_arn role_session_name web_identity_token_file].freeze
+
+    def initialize(params)
+      @params = params
+    end
+
+    # Returns either IAM or STS credentials to be used on calls to S3
+    #
+    # @param [Hash] Hash containing AWS credentials related keys
+    # @return [Hash<Symbol, String>] with access_key_id and secret_access_key when IAM credentials are provided
+    # @return [Hash<Symbol, Aws::AssumeRoleWebIdentityCredentials>] when STS credentials are provided
+    # @raise [AuthenticationTypeError] if not enough params are provided
+    def call
+      if valid_params_with(iam_params, IAM_KEYS)
```

Comment:
Shouldn't STS be the default one, and use IAM only when no STS is provided? that's what the [comment](https://github.com/3scale/porta/pull/3066/files#r980422584) above suggests, but the PR description and this code says the opposite.

Reply:
Using IAM is something more manual than the automatic STS provided by operator. At least this is my understanding. So to me it makes more sense to give priority to the things users manually do/configure (i.e. IAM with priority).

Reply:
makes sense